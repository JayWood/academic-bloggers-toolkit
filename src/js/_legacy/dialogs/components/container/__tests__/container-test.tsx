import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import Container from '..';

const setup = () => {
    const onClose = jest.fn();
    const component = shallow(
        <Container title="Hello World" onClose={onClose}>
            <h1>Hello World</h1>
        </Container>,
    );
    return {
        component,
        instance: component.instance() as Container,
        onClose,
    };
};

describe('<Container />', () => {
    it('should match snapshot', () => {
        const { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should render', () => {
        const { component } = setup();
        expect(component).toBeTruthy();
    });
    it('should handle mouse wheel appropriately', () => {
        const { component } = setup();
        const preventDefault = jest.fn();
        component.simulate('wheel', { cancelable: false, preventDefault });
        expect(preventDefault).not.toHaveBeenCalled();
        component.simulate('wheel', { cancelable: true, preventDefault });
        expect(preventDefault).toHaveBeenCalledTimes(1);
    });
    it('should close with and without an associated event', () => {
        const { instance } = setup();
        expect(instance.close).not.toThrow();
        const preventDefault = jest.fn();
        instance.close({ preventDefault } as any);
        expect(preventDefault).toHaveBeenCalled();
    });
    it('should handle key event', () => {
        const { instance, onClose } = setup();
        const mockEvent = {
            key: 'a',
            stopPropagation: jest.fn(),
        };
        (instance as any).handleKeyEvent(mockEvent);
        mockEvent.key = 'Escape';
        (instance as any).handleKeyEvent(mockEvent);
        expect(onClose).toHaveBeenCalledTimes(1);
        expect(mockEvent.stopPropagation).toHaveBeenCalledTimes(1);
    });
    it('should toggle focus trap', () => {
        const { instance } = setup();
        expect(instance.focusTrapPaused).toBe(false);
        instance.toggleFocusTrap();
        expect(instance.focusTrapPaused).toBe(true);
    });
});
