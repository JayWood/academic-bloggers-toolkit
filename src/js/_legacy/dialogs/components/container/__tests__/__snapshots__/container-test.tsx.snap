// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div
  className="container"
  onKeyDown={[Function]}
  onWheel={[Function]}
  style={
    Object {
      "background": "rgba(0, 0, 0, 0.7)",
    }
  }
>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
    tag="div"
  >
    <div
      aria-labelledby="dialog-label-Hello_World"
      className="dialog"
      role="dialog"
      style={
        Object {
          "position": "relative",
          "width": 600,
        }
      }
    >
      <header
        className="header"
      >
        <span
          className="dialogTitle"
          id="dialog-label-Hello_World"
        >
          Hello World
        </span>
        <button
          aria-label="Close dialog"
          className="button"
          onClick={[Function]}
        >
          <span
            className="dashicons dashicons-no-alt"
          />
        </button>
      </header>
      <h1
        toggleFocusTrap={[Function]}
      >
        Hello World
      </h1>
    </div>
  </FocusTrap>
</div>
`;
